aura/lib/Aura/Settings/External.hs

```diff
@@ -45,17 +46,28 @@ data AuraConfig = AuraConfig
   , acAnalyse   :: !(Maybe BuildSwitch) }
   deriving stock (Show)
 
-defaultAuraConf :: FilePath
-defaultAuraConf = "/etc/aura.conf"
+userAuraConfPath :: IO FilePath
+userAuraConfPath = getUserConfigFile "aura" "config"
 
-getAuraConf :: FilePath -> IO Config
-getAuraConf fp = do
-  exists <- doesFileExist fp
+systemAuraConfPath :: FilePath
+systemAuraConfPath = "/etc/aura.conf"
+
+-- | Attempt to get a valid Aura config from a specified path.
+getAuraConfFrom :: FilePath -> IO (Maybe Config)
+getAuraConfFrom path = do
+  exists <- doesFileExist path
   if not exists
-    then pure $ Config mempty
+    then pure Nothing
     else do
-      file <- decodeUtf8Lenient <$> BS.readFile fp
-      pure . either (const $ Config M.empty) id $ parse config "aura config" file
+      file <- decodeUtf8Lenient <$> BS.readFile path
+      pure . hush $ parse config "aura config" file
+
+getAuraConf :: IO Config
+getAuraConf = do
+  userCfg <- getAuraConfFrom =<< userAuraConfPath
+  case userCfg of
```

Comment:
Since this `userCfg` symbol is only used in one place, do you want to try collapsing this with the previous line using [`LambdaCase`](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html?highlight=derivingvia#extension-LambdaCase)?

Reply:
Pushed a change, is that what you had in mind?

I've never thought of combining bind and lambdacase like this!

Reply:
Yes it's a nice trick that really cleans things up.